File: 프로그래머스/lv1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    arr = [1 for x in range(n)]
    for i in lost:
        arr[i-1] -=1
    for r in reserve:
        arr[r-1] +=1
    for i in range(n):
        if arr[i]==2:
            if i-1>=0 and arr[i-1]==0:
                arr[i-1]=1
                arr[i]=1
            elif i+1<n and arr[i+1]==0:
                arr[i+1]=1
                arr[i]=1
            
    cnt=0
    for i in arr:
        if i>=1:
            cnt+=1
    return cnt

File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    let arr=Array(n).fill(1)
    for(let l of lost) arr[l-1]--
    for(let r of reserve) arr[r-1]++
    for(let i=0;i<arr.length;i++){
        if(arr[i]===0){
            if(arr[i-1]>1){
                arr[i-1]--
                arr[i]++
            } else if(arr[i+1]>1){
                arr[i+1]--
                arr[i]++
            }
        }
    }
    let cnt=0
    for(let i=0;i<arr.length;i++) {
        if(arr[i]===0) cnt++
    }
    return arr.length-cnt
}